### PR TITLE
:penguin: introduced grunt-task that creates an archlinux package

### PR DESCRIPTION
**Pull request by wampiedriessen**
This needs some reviewing, for I am just a regular arch user. Not a guru of sorts :)

### CODE REVIEW

**Comment by paulcbetts** [comment]
@wampiedriessen Awesome! Thanks for this

File: script/mkpkg [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+# mkdeb name version channel arch control-file-path desktop-file-path icon-path pkg-file-path
```
Comment (paulcbetts) [inline script/mkpkg]: mkpkg

**Comment by wampiedriessen** [comment]
Can the Travis check be re-run? Somehow, I don't believe my changes could have caused electron to be unable to be dowloaded

**Comment by 50Wliu** [comment]
Restarted!

File: resources/linux/archlinux/PKGBUILD.in [hunk 2]
```diff
@@ -0,0 +1,16 @@
+pkgname=<%= appFileName %>
+pkgver=<%= version %>
+pkgrel=1
+pkgdesc="Chrome-based text editor from Github"
```
Comment (mehcode) [inline resources/linux/archlinux/PKGBUILD.in]: The hackable text editor for the 21st century, built on Electron.

File: resources/linux/archlinux/PKGBUILD.in [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+arch=('<%= arch %>')
+url="http://atom.io"
+license=('MIT')
+provides=('atom')
+conflicts=('atom-editor' 'atom-editor-bin')
+depends=('nss' 'gtk2' 'libnotify' 'alsa-lib' 'gconf' 'libxtst' 'nodejs' 'libgnome-keyring' 'python' 'desktop-file-utils')
+makedepends=('npm')
+md5sums=()
```
Comment (mehcode) [inline resources/linux/archlinux/PKGBUILD.in]: I wouldn't list `nodejs` or `npm` as a dependency. As far as I know it isn't as atom packages it's own copy of node and npm. 

Actually.. we should just use the depends from `atom-editor-bin`:

```
depends=('git' 'gconf' 'gtk2' 'libnotify' 'libxtst' 'nss' 'python2' 'xdg-utils' 'desktop-file-utils' 'alsa-lib' 'libgnome-keyring')
optdepends=('gvfs')
```

And you don't need `npm` as a `makedepends` as atom is already built. We just need `cp` here.

File: resources/linux/archlinux/PKGBUILD.in [hunk 4]
```diff
@@ -0,0 +1,16 @@
+pkgname=<%= appFileName %>
+pkgver=<%= version %>
+pkgrel=1
+pkgdesc="Chrome-based text editor from Github"
+arch=('<%= arch %>')
+url="http://atom.io"
+license=('MIT')
+provides=('atom')
+conflicts=('atom-editor' 'atom-editor-bin')
```
Comment (mehcode) [inline resources/linux/archlinux/PKGBUILD.in]: ```conflicts=('atom-editor' 'atom-editor-git' 'atom-editor-git-tagged atom-editor-bin')```

File: resources/linux/archlinux/PKGBUILD.in [hunk 5]
```diff
@@ -0,0 +1,16 @@
+pkgname=<%= appFileName %>
+pkgver=<%= version %>
+pkgrel=1
+pkgdesc="Chrome-based text editor from Github"
+arch=('<%= arch %>')
+url="http://atom.io"
+license=('MIT')
+provides=('atom')
```
Comment (mehcode) [inline resources/linux/archlinux/PKGBUILD.in]: The `provides` line isn't doing anything useful.

File: resources/linux/archlinux/PKGBUILD.in [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+conflicts=('atom-editor' 'atom-editor-bin')
+depends=('nss' 'gtk2' 'libnotify' 'alsa-lib' 'gconf' 'libxtst' 'nodejs' 'libgnome-keyring' 'python' 'desktop-file-utils')
+makedepends=('npm')
+md5sums=()
+
+package() {
+  cp -r * "$pkgdir/"
```
Comment (mehcode) [inline resources/linux/archlinux/PKGBUILD.in]: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=atom-editor-bin#n26 

There is code (in atom-editor-bin) to use a python patch so atom is forced to use python2 internally. Can atom work with python3 as the default python?

**Comment by damieng** [comment]
Sorry - we have recently rewritten our build system and it no longer uses grunt which invalidates this PR.

**Pull request closed without merging**